### DISCUSSION TITLE
Liquify / Distortions - And keyframing them!

### DISCUSSION THREAD

**Discussion by GreatGravy** [keiwando/toonsquid#50] [Ideas]
Loving the app so far, but some sort of way to animate using liquify or another distortion would be amazing - especially with the vector tools you already have. I love making a base illustration and modifying it as a way to animate it rather than redrawing the frame from scratch, it's mostly how I currently animate in procreate. However, I think introducing keyable liquifies or distortions would be extremely powerful for animation, especially since in procreate I would have to destructively liquify/warp the image, and eyeball it for every single frame - being able to key these modifications would be a lifechanger for me!

**Comment by AQuickSecond** [comment]
Sounds interesting, could be surprisingly versatile. I would most likely use these tools.

**Comment by SkipperWIng** [comment]
Not sure this is in the same vein, but yeah, I do the same thing between keys for easing, so I don't have to redraw every single frame. Having some kind of "Skew/Distort" functionality (as an alternative inside the Transform Tool, similar to the "Uniform" and "Freeform" options inside the "Scaling Mode") would be super helpful. 
Something where the top/side/bottom anchors control skew (or simply the border edges), but the corners control distortion?
And agree, warp functionality would be awesome as well (if it is feasible; I haven't seen it in either Rough Animator or Callipeg which makes me question how feasible it is in these apps)
In any case, this felt like the closest thread to what I was looking for, so I felt it was tidier to post here, rather than start a new thread...?

**Comment by keiwando** [comment]
Perspective and Warp transform options are now available for pixel layers with the 1.7 update.
Making them animatable using keyframes will be a separate feature for me to look into at some point in the future.
Thanks for the feedback!
